src/NServiceBus.Transport.RabbitMQ.TransportTests/ConfigureRabbitMQTransportInfrastructure.cs

```diff
@@ -55,7 +46,38 @@ public Task Cleanup()
 
     void PurgeQueues()
     {
-        var connectionFactory = CreateConnectionFactory();
+        if (connectionStringBuilder == null)
```

Comment:
I'm not sure this is enough of a check to avoid all problems. If the connection string exists, but it fails validation at line 28, then `Cleanup` will still be called when the test didn't run, and we had some problem with the connection string we'll try to repeat here.